### ISSUE TITLE
No property checking done when appending to array

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#105]
```php
class A {
    /** @var string[] */
    public $strs = ["a", "b", "c"];

    public function foo() {
        $this->strs = [new stdClass()]; // InvalidPropertyAssignment issue correctly emitted
    }
  
    public function bar() {
        $this->strs[] = new stdClass(); // no issue emitted
    }
}
```
